src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -879,13 +882,35 @@ export class MatAutocompleteTrigger
     });
   }
 
+  /** Implements BreakpointObserver to be used to detect handset landscape */
+  private _breakpointObserver = inject(BreakpointObserver);
+
   private _getOverlayPosition(): PositionStrategy {
+    // Set default Overlay Position
     const strategy = this._overlay
       .position()
       .flexibleConnectedTo(this._getConnectedElement())
       .withFlexibleDimensions(false)
       .withPush(false);
 
+    if (!this._handsetLandscapeBreakpointSubscription) {
+      // Subscribe to the breakpoint events stream to detect when screen is in
+      // handsetLandscape. Only subscribe if/when this panel is open.
+      // BreakpointObserver only returns screen size or isMatched/matches boolean.
+      this._handsetLandscapeBreakpointSubscription = this._breakpointObserver
+        .observe(Breakpoints.HandsetLandscape)
+        .subscribe(result => {
+          // Check breakpoint if being viewed in HandsetLandscape via subscription (if not null)
+          const isHandsetLandscape = result.matches;
+          // Apply HandsetLandscape settings to prevent overlay cutoff in that breakpoint
+          // Fixes b/284148377
+          if (isHandsetLandscape) {
+            strategy.withFlexibleDimensions(true).withGrowAfterOpen(true).withViewportMargin(8);
```

Comment:
do we need to unset these options if it changes back to portrait?

Reply:
Updated to set to previous default settings if isHandsetLandscape is false.